### DISCUSSION TITLE
X1-Hybrid-G4 Modbus Connection

### DISCUSSION THREAD

**Discussion by jameski83** [wills106/homeassistant-solax-modbus#36] [Q&A]
Hi,

I am trying to get the data via ethernet RS485 (connected to COM port on the inverter) to USB connected to a raspberry PI, then to Homeassistant which is hosted in a VM in another location in my house but not having much success, is this possible? Any help would be greatly appreciated.

**Comment by infradom** [comment]
Do you meen you are using a modbus-RS485 to modbus over Ethernet converter? If so, which brand?
I myself have used a 2€ usb to RS485 converter to connect my G4 to Homeassistant on a Raspberry
Afaik, the native Ethernet connection of the Gen4 inverters do not carry any modbus tinterface, so you either need a Eternet-to-Rs485 or usb-to-RS485 adapter

**Reply by jameski83** [reply]
I'm using a USB to RS485 cable, connected to the com port on my inverter which is RJ45. I have the pocket lan adapter but as you say it doesn't expose modbus data. 

Maybe I've got the wrong cable? It's Future Technology Devices when I run lsusb

**Comment by infradom** [comment]
Polarity does matter if I remember well, I had to swap the wires.
Did you connect the blue/white pair of the utp cable to the USB adapter ?

Jan

On Wed, Mar 9, 2022 at 6:08 PM jameski83 ***@***.***> wrote:

> I'm using a USB to RS485 cable, connected to the com port on my inverter
> which is RJ45. I have the pocket lan adapter but as you say it doesn't
> expose modbus data.
>
> Maybe I've got the wrong cable? It's Future Technology Devices when I run
> lsusb
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/wills106/homsassistant-solax-modbus/discussions/36#discussioncomment-2325705>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC2B23SGB34WGDGT7TOCUMLU7DLHXANCNFSM5QJ7C5UQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you commented.Message ID:
> <wills106/homsassistant-solax-modbus/repo-discussions/36/comments/2325705@
> github.com>
>


-- 
Jan Colpaert
+32 478 882435

**Reply by jameski83** [reply]
It's a sealed cable I didn't have to manually connect any wires. 

https://www.amazon.co.uk/dp/B0881DYLY6/ref=cm_sw_r_cp_api_glt_i_XRW1GA2KTPRFMHA34H05?_encoding=UTF8&psc=1

**Comment by infradom** [comment]
I am using this device:
https://www.berrybase.de/raspberry-pi/raspberry-pi-computer/usb-geraete/usb-rs485-konverter

It uses the CH340 chip

Polarity:
[image: image.png]

On Wed, Mar 9, 2022 at 6:08 PM jameski83 ***@***.***> wrote:

> I'm using a USB to RS485 cable, connected to the com port on my inverter
> which is RJ45. I have the pocket lan adapter but as you say it doesn't
> expose modbus data.
>
> Maybe I've got the wrong cable? It's Future Technology Devices when I run
> lsusb
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/wills106/homsassistant-solax-modbus/discussions/36#discussioncomment-2325705>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC2B23SGB34WGDGT7TOCUMLU7DLHXANCNFSM5QJ7C5UQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you commented.Message ID:
> <wills106/homsassistant-solax-modbus/repo-discussions/36/comments/2325705@
> github.com>
>


-- 
Jan Colpaert
+32 478 882435

**Reply by jameski83** [reply]
Maybe I should return my cable and get that adapter and manually connect it if you have the same inverter as me and it works

**Comment by infradom** [comment]
The picture is not clear, but are you sure it uses the middle 2 pins on the
RJ45 plug (pin 4 and 5) ?

Jan

On Wed, Mar 9, 2022 at 6:23 PM jameski83 ***@***.***> wrote:

> It's a sealed cable I didn't have to manually connect any wires.
>
>
> https://www.amazon.co.uk/dp/B0881DYLY6/ref=cm_sw_r_cp_api_glt_i_XRW1GA2KTPRFMHA34H05?_encoding=UTF8&psc=1
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/wills106/homsassistant-solax-modbus/discussions/36#discussioncomment-2325806>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC2B23WF3CUMJJ2NDVTUJ43U7DNBJANCNFSM5QJ7C5UQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you commented.Message ID:
> <wills106/homsassistant-solax-modbus/repo-discussions/36/comments/2325806@
> github.com>
>


-- 
Jan Colpaert
+32 478 882435

**Reply by jameski83** [reply]
Yes I've just checked and it does 100%

**Comment by infradom** [comment]
The used chipset should be fine (even better) but the pinout might be wrong
.

Jan

On Wed, Mar 9, 2022 at 6:26 PM jameski83 ***@***.***> wrote:

> Maybe I should return my cable and get that adapter and manually connect
> it if you have the same inverter as me and it works
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/wills106/homsassistant-solax-modbus/discussions/36#discussioncomment-2325822>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC2B23TU5E6ZAZ5IMNXRKOTU7DNNTANCNFSM5QJ7C5UQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you commented.Message ID:
> <wills106/homsassistant-solax-modbus/repo-discussions/36/comments/2325822@
> github.com>
>


-- 
Jan Colpaert
+32 478 882435

**Comment by infradom** [comment]
Did you leave the communication speed of the solax inverter at 19200 ?
In the setup dialogue, I specify the usb adapter by id:
/dev/serial/by_id/xxxxx
It should also work with /dev/ttyUSB0  (or whatever port you use)




On Wed, Mar 9, 2022 at 6:31 PM jameski83 ***@***.***> wrote:

> Yes I've just checked and it does 100%
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/wills106/homsassistant-solax-modbus/discussions/36#discussioncomment-2325847>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC2B23SOIXOJFALTPHSAV4DU7DN53ANCNFSM5QJ7C5UQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you commented.Message ID:
> <wills106/homsassistant-solax-modbus/repo-discussions/36/comments/2325847@
> github.com>
>


-- 
Jan Colpaert
+32 478 882435

**Reply by jameski83** [reply]
It is /dev/ttyUSB0 but Homeassistant is not on the raspberry pi it's on a VM elsewhere in my house

**Comment by infradom** [comment]
Sorry, did not realize that. There must be a way to tunnel the serial
protocol over a tcp socket on the raspberry, but I cannot immediately help
here at the moment, too many other things to do...


On Wed, Mar 9, 2022 at 6:39 PM jameski83 ***@***.***> wrote:

> It is /dev/ttyUSB0 but Homeassistant is not on the raspberry pi it's on a
> VM elsewhere in my house
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/wills106/homsassistant-solax-modbus/discussions/36#discussioncomment-2325909>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC2B23WHEFORPCCRPQN5WADU7DO6DANCNFSM5QJ7C5UQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you commented.Message ID:
> <wills106/homsassistant-solax-modbus/repo-discussions/36/comments/2325909@
> github.com>
>


-- 
Jan Colpaert
+32 478 882435

**Reply by infradom** [reply]
maybe this will help 
 https://raspberrypi.stackexchange.com/questions/15382/raspberry-pi-used-as-a-cheap-serial-to-wlan-converter

**Reply by wills106** [reply]
Another option? Never tried it myself https://github.com/custom-components/remote_homeassistant

**Comment by wills106** [comment]
Are you not able to run Cat5 from the Inverter to the USB Adapter directly connected to your VM?
RS485 over Cat5 is meant to be able to run up to 1200m.
Then you can bypass the Raspberry PI completely.

**Reply by jameski83** [reply]
I'm going to try installing homeassistant on the pi and test with a direct connection to check it is outputting data as I've tried to do it over the network and it doesn't get any data. A direct cable connection is possible but I thought it would be easier if I could use my existing network first

**Reply by jameski83** [reply]
it works fine directly connected

**Comment by infradom** [comment]
furthermore, you have to make sure that the RS485 port is enabled in the
solax advanced settings



On Wed, Mar 9, 2022 at 6:31 PM jameski83 ***@***.***> wrote:

> Yes I've just checked and it does 100%
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/wills106/homsassistant-solax-modbus/discussions/36#discussioncomment-2325847>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC2B23SOIXOJFALTPHSAV4DU7DN53ANCNFSM5QJ7C5UQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you commented.Message ID:
> <wills106/homsassistant-solax-modbus/repo-discussions/36/comments/2325847@
> github.com>
>


-- 
Jan Colpaert
+32 478 882435
